src/hotspot/share/jfr/support/jfrThreadLocal.hpp

```diff
@@ -342,6 +367,45 @@ class JfrThreadLocal {
   void set_thread_blob(const JfrBlobHandle& handle);
   const JfrBlobHandle& thread_blob() const;
 
+  // CPU time sampling
+#ifdef LINUX
+  void set_cpu_timer(timer_t timer) {
+    _has_cpu_timer = true;
+    _cpu_timer = timer;
+  }
+
+  void unset_cpu_timer() {
+    _has_cpu_timer = false;
+  }
+
+  timer_t cpu_timer() const {
+    return _cpu_timer;
+  }
+
+  bool has_cpu_timer() const {
+    return _has_cpu_timer;
+  }
+
+  bool is_cpu_time_jfr_enqueue_locked() { return Atomic::load(&_cpu_time_jfr_locked) == ENQUEUE; }
+  bool is_cpu_time_jfr_dequeue_locked() { return Atomic::load(&_cpu_time_jfr_locked) == DEQUEUE; }
+
+  bool acquire_cpu_time_jfr_enqueue_lock();
+  bool acquire_cpu_time_jfr_native_lock();
+  void acquire_cpu_time_jfr_dequeue_lock();
+  void release_cpu_time_jfr_queue_lock();
+
+  void set_has_cpu_time_jfr_requests(bool has_events);
+  bool has_cpu_time_jfr_requests();
+
+  JfrCPUTimeTraceQueue& cpu_time_jfr_queue();
+  void disable_cpu_time_jfr_queue();
```

Comment:
Seems a bit strange that you can "disable" a queue. When is it enabled? Since the underlying implementation delegates to ensure_capacity for size == 0, and deallocates, perhaps a more appropriate name is "deallocate_cpu_time_jfr_queue()"?